ext/neural_snow/NeuralSnow.jl

```diff
@@ -0,0 +1,240 @@
+    zmodel_state = BSON.load(IOBuffer(HTTP.get(download_link).body))[:zstate]
+    Flux.loadmodel!(zmodel, zmodel_state)
+    ModelTools.settimescale!(zmodel, 86400.0)
+    return zmodel
+
+    #should I write this function to give the ability to also load from local file too?
+    #BSON.@load "../SnowResearch/cleancode/testhrmodel.bson" zstate
+    #Flux.loadmodel!(zmodel, zstate)
+end
+
+"""
+    converted_model_type!(model::Flux.Chain, FT::Type)
+A wrapper function to aid conversion of the utilized Flux network weights to the type determined by the simulation.
+Currently supports `Float32` and `Float64` types.
+"""
+function converted_model_type(model::Flux.Chain, FT::DataType)
+    if FT == Float32
+        return Flux.f32(model)
+    elseif FT == Float64
+        return Flux.f64(model)
+    else
+        error(
+            "Conversion of Flux Chain weights to the desired float-type is not supported. Please implement the desired conversion
+      in NeuralSnow.convert_model_type!() or a custom constructor for the NeuralDepthModel with the desired type.",
+        )
+    end
+end
+
+"""
+    NeuralDepthModel(FT::DataType; Δt::Union{Nothing, FT}=nothing; model::Flux.Chain, α::Union{FT, Nothing})
+An outer constructor for the `NeuralDepthModel` density parameterization for usage in a snow model.
+Can choose to use custom models via the `model` argument, or a custom exponential moving average weight for the network inputs.
+The Δt argument can be used to set the lower boundary value on the network (when not using a custom model) in accordance with the paper upon initialization,
+equivalent to ModelTools.settimescale!(model, Δt, dtype = FT).
+"""
+function NeuralDepthModel(
+    FT::DataType;
+    Δt = nothing,
+    model::Flux.Chain = get_znetwork(),
+    α = nothing,
+)
+    usemodel = converted_model_type(model, FT)
+    weight =
+        !isnothing(α) ? FT(α) :
+        (isnothing(Δt) ? FT(1 / 12) : FT(maximum([2 * Δt / 86400, 1])))
+    # should we set the timescale of the model to Δt? getting rid of get_znetwork() and below line mitigates the need to import ModelTools entirely.
+    if !isnothing(Δt)
+        ModelTools.settimescale!(usemodel, Δt, dtype = FT) #assumes Δt is provided in seconds
+    end
+    return NeuralDepthModel{FT}(usemodel, weight)
+end
+
+#Define the additional prognostic variables needed for using this parameterization:
+#do I need to export these for them to be utilized in the simulation?
+ClimaLand.Snow.density_prog_vars(m::NeuralDepthModel) =
+    (:Z, :P_avg, :T_avg, :R_avg, :Qrel_avg, :u_avg)
+ClimaLand.Snow.density_prog_types(m::NeuralDepthModel{FT}) where {FT} =
+    (FT, FT, FT, FT, FT, FT)
+ClimaLand.Snow.density_prog_names(m::NeuralDepthModel) =
+    (:surface, :surface, :surface, :surface, :surface, :surface)
+
+#so far, every type is just blank here - is it even worth providing this feature or just get rid of this feature in Snow.jl entirely?
+ClimaLand.Snow.density_aux_vars(m::NeuralDepthModel) = ()
```

Comment:
you can define a single default method in Snow.jl, e.g.
ClimaLand.Snow.density_aux_vars(m::AbstractDensityModel) = ()

and then not define it here - the NeuralDepthModel would fall back on the default if it does not have a more specific method

That said, it *none* of the density models need additional vars, you can just not make this function at all